extensions/2.0/Khronos/KHR_audio_graph/README.md

```diff
@@ -0,0 +1,1279 @@
+   </td>
+  </tr>
+  <tr>
+   <td>auto play
+   </td>
+   <td>boolean
+   </td>
+   <td>Play on load.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>loop
+   </td>
+   <td>boolean
+   </td>
+   <td>Playback in a loop (false = no-loop, true = loop). If set to true, then once playback reaches the time specified by “loop end position” (or the end of the asset, whichever is first), the source node will continue playback again from a position specified by the “loop start position” property.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>position
+   </td>
+   <td>number
+   </td>
+   <td>Play position in ms.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>gain
+   </td>
+   <td>number
+   </td>
+   <td>Gain value applied to the audio.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>playback speed
+   </td>
+   <td>number
+   </td>
+   <td>Rate of playback. A value of 1.0 would playback the audio at the standard rate. A value of 2.0 would play back the asset at double the speed.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>loop start position
+   </td>
+   <td>number
+   </td>
+   <td>The starting position in ms when looping.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>loop end position
+   </td>
+   <td>number
+   </td>
+   <td>The ending position (ms) of the loop.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>duration
+   </td>
+   <td>number
+   </td>
+   <td>Length of the underlying audio data in ms.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>encoding properties
+   </td>
+   <td>object
+   </td>
+   <td>See 4.4 Encoding properties.
+   </td>
+   <td>
+   </td>
+   <td>Could be a part of audio data property  instead.
+   </td>
+  </tr>
+</table>
+
+
+
+### 4.2 Audio data
+
+Audio data objects define where audio data is located and what format the data is in. The data is either accessed via a bufferView or uri.
+
+
+<table>
+  <tr>
+   <td>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Notes</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>buffer view
+   </td>
+   <td>integer
+   </td>
+   <td>The index of the buffer view that contains the audio data. The buffer represents an audio asset residing in memory, created from decoding an audio file, or from raw data.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>mime type
+   </td>
+   <td>string
+   </td>
+   <td>The audio's MIME type. Required if buffer view is defined.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>uri
+   </td>
+   <td>string
+   </td>
+   <td>The uri of the audio file. Relative paths are relative to the .gltf file.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+
+### 4.3 Oscillator data
```

Comment:
Has band-limiting been considered for audio rate oscillator implementations? Is it assumed? Or does it need to be explicitly required? I would think it should be explicitly required if audio quality is a priority.

Reply:
Made that explicit by including a detune parameter. The implementation can then use this with frequency to calculate the desired oscillator frequency.

Reply:
Ok, thanks, but I don't think adding a `detune` parameter has anything to do with making band-limiting explicit. If it's assumed that implementations will apply band-limiting to avoid aliasing, then that's fine, but I suspect many implementations will take the easier route of skipping band-limiting if it's not required, which will reduce audio quality.

It may be that this concern is outside the scope of the standard, but if there a lot of poor-quality implementations then it may cause creators to avoid using it, so we might want to make band-limiting required.

Reply:
There are various practical methods an implementation might use to prevent aliasing. While the ideal discrete-time digital audio signal is clearly defined mathematically, each implementation must balance the trade-off between computational cost and fidelity to this ideal. It is anticipated that implementations will strive towards this ideal, but it is also reasonable to consider lower-quality, more cost-effective approaches for less powerful hardware. We can include more guidance/recommendation around this.